### PR TITLE
ES-5 compatible mappings and related changes

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Product Description

This PR updates the mappings which are compatible with ES 5. The first commit https://github.com/dimagi/commcare-hq/pull/33485/commits/355bf7fda181089f2c4a8bab361a63df6328a1a2 is used to transform the mappings.
The mappings are transformed as per logic - 
```
string + not_analyzed => keyword
string => text
```
The commits after that are just result of the transformation applied on each of the mapping file.

Then the next 2 commits are small updates in mappings that is done by going through the changelog.

After that the commits update the test suite to run against ES 5.

Review by commit 🐟 

## Technical Summary

https://dimagi-dev.atlassian.net/browse/SAAS-14891
### Safety story

Changes just affect the test suite and ensures that tests run against ES5 .
### Automated test coverage

NA
### QA Plan

NA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: docker/localsettings.py [hunk 1]
```diff
@@ -100,12 +100,11 @@
...
-ELASTICSEARCH_HOST = 'elasticsearch2'
-ELASTICSEARCH_PORT = 9200  # ES 2 port
-ELASTICSEARCH_MAJOR_VERSION = 2
+ELASTICSEARCH_HOST = 'elasticsearch5'
+ELASTICSEARCH_PORT = 9200  # ES 5 port
+ELASTICSEARCH_MAJOR_VERSION = 5
 
-if os.environ.get('ELASTICSEARCH_MAJOR_VERSION'):
-    ELASTICSEARCH_MAJOR_VERSION = int(os.environ.get('ELASTICSEARCH_MAJOR_VERSION'))
+ELASTICSEARCH_MAJOR_VERSION = 5
```
Comment (millerdev) [inline docker/localsettings.py:107]: Can this duplicate line be removed?

**Review by millerdev — COMMENTED** [review]
How does one decide to use `text` or `keyword` when converting from `"type": "string"`?

**Comment by AmitPhulera** [comment]
> How does one decide to use text or keyword when converting from "type": "string"?

@millerdev I had added those details in the transform function that I created in 355bf7fda181089f2c4a8bab361a63df6328a1a2 but while cherry-picking it was missed. 
I have added it back in the PR

File: docker/localsettings.py (hunk 1 above)
Comment (AmitPhulera) [reply docker/localsettings.py]: [945dfd8](https://github.com/dimagi/commcare-hq/pull/33485/commits/945dfd8a6dd1adcc4219e91318563c2dad305478)

File: docker/localsettings.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-####### Configuration for CommCareHQ Running in docker #######
+"""####### Configuration for CommCareHQ Running in docker #######"""
```
Comment (millerdev) [inline docker/localsettings.py:1]: Interesting solution to that lint error. Would it also work to prefix the line with `# `?
```suggestion
# ####### Configuration for CommCareHQ Running in docker #######
```
Not saying you should change it, just trying to think of another solution.

**Review by millerdev — APPROVED** [review]

File: docker/localsettings.py (hunk 2 above)
Comment (AmitPhulera) [reply docker/localsettings.py:1]: Yeah it will work with `#  `, I tried that too but did not like the extra space in the beginning 😅

**Pull request merged**